radioapp/core/service_discovery.py

```diff
@@ -1,11 +1,42 @@
 """
 radio.core.service_discovery
 ============================
-
 The Service Discovery submodule is responsible for locating and identifying
 backend instances. In normal operation, it must find the address of the backend
 that was configured in the provisioning mode. In provisioning mode, it must
 discover and return the names of all available backend instances present on the
 local network. This will accomplished via a standard service discovery protocol,
 multicast DNS (mDNS).
-"""
\ No newline at end of file
+"""
+
+from zeroconf import ServiceBrowser, Zeroconf
+import logging
+
+class BackendDiscovery:
```

Comment:
Have you tested this? (Run avahi-discover, pick out a service type that's visible on your network, make sure this gets the same results).